### ISSUE TITLE
Duplicate replies are being made sometimes

### ISSUE THREAD

**Issue by orangespire** [kyb3r/modmail#401]
I have noticed with some threads when I try to reply, It creates two replies and the user never gets the message sent.  I have tried some of the troubleshooting steps mentioned such as generating a new bot token so It's not a another instance of the bot running.  I have also tried restarting the bot.  I have checked the logs and it does not show anything (https://hasteb.in/apunopix) related to this issue.  

Here is a screenshot of the issue for reference: ![](https://i.imgur.com/6A8BTSY.png)

**Comment by u187** [comment]
I can confirm this is an issue. Here are some details with the setup I use:
- Hosted with Heroku (bot + logviewer)
- Separate server setup (this may be where the issues lay as we had no problems with the modmail stuff on the same server)
- Have reset and double checked the config vars a few times to make sure everything is correct.
- Have removed the "MODMAIL_GUILD_ID" and set the separate modmail server as the main "GUILD_ID", as suggested by staff on the Discord server. This does nothing and doesn't solve the issue.
- The bot is only in two servers, the main server, and the separate modmail server.

When this issue occurs to a thread, responses will not be relayed to either party (dm or the channel).  The person who opened a modmail thread will not be able to open another one until the original one is manually closed and essentially won't know their messages aren't seen. We've been using ?close and then ?contact to fix the issue, but the issue basically happens with every thread eventually after a while. I wonder if it has something to do with the bot restarting when there are open threads? As I noticed the uptime to the bot was 5 hours ago and the bugged open thread was opened 12 hours ago, so when responding the duplicate messages bug happened. Multiple people have reported this on the discord, hopefully, it can be fixed.

**Comment by Taaku18** [comment]
@u187 Please open a new issue as the problem you stated is different from the original. From my understanding, the original question was bot responding twice and yours is broken/unresponsive thread.

If there’s actually more users encountering the same problem to any of your questions, please let them post their version of the logs and problem so we can attempt to diagnose it since these are unreproducible issues. But to my knowledge, only you two have ever reported an issue on the Discord server; however, I’ll leave this issue open for any additional users, if any, can comment.


The original questions common causes:
  - Two bots running with the same token
  - Extreme client lag (highly unlikely)
  - Modified Modmail bot

The latter questions common causes:
  - Lack of bot permissions (read message history)
  - ???

**Comment by u187** [comment]
It's actually the same issue, I'm just adding on to what the original poster submitted. It makes more sense to have it all on one issue instead of creating multiple duplicate issues for the exact same thing no?

**Comment by StephenDaDev** [comment]
Some users have been able to resolve this issue by updating their bot, and restarting it. It has also been seen to be fixed by simply regening the token and replacing the config var for it.

**Comment by StephenDaDev** [comment]
Interesting, just had this happen to in our other support channels, only after around 24 hours did this happen, did this happen BEFORE you changed it to a sep server setup?

**Comment by u187** [comment]
@StephenDaDev — not that I noticed, we only started noticing the problem after we separated the logs on another server.

**Comment by StephenDaDev** [comment]
@Taaku18, In that case, its a bug with the sep server setup 
```To use a separate server as the inbox for threads, set the following config variables:

- MODMAIL_GUILD_ID (the server where messages are sent to)
- GUILD_ID (The server where users message from)
```
Multiple reported incidents, the last person was able to fix by removing it and setting the modmail to inbox server ONLY

**Comment by StephenDaDev** [comment]
I have tried to reproduce this multiple times on multiple different hosting platforms, on a VPS I got 0 out of 2 doing it on heroku I got 1 of 1 I'm going to keep messing around with this to see if the sep server actually has an issue.

**Comment by officialpiyush** [comment]
Can confirm this issue, happened to me today

**Comment by wc3717** [comment]
Same issue, kicked it from one server and now it is sending double replies.

**Comment by StephenDaDev** [comment]
Hey Will! Are you using Separate Server setup or no?

**Comment by wc3717** [comment]
> Hey Will! Are you using Separate Server setup or no?

@StephenDaDev Yes we are, we have the general server and then a staff server.

**Comment by StephenDaDev** [comment]
Ok, interesting, its the Windows VPS with 32gb of RAM or the linux one?

**Comment by catsarecoo** [comment]
wew I made it over the waves ~

**Comment by wc3717** [comment]
It also appears as double messages are not sending at all to the user.

**Comment by StephenDaDev** [comment]
Yeah, this appears to be an even larger issue than the other instances we are currently dealing with. I have a feeling that it has something to do with Matrix/Andy's changes, as the problem is more severe than any other instance.

**Comment by officialpiyush** [comment]
> I have a feeling that it has something to do with Matrix/Andy's changes, as the problem is more severe than any other instance.

Can you link the changes, please? I just checked matrix's [commits](https://github.com/kyb3r/modmail/commits?author=matrix2113) and I couldn't find any commit which alters core features in any way and would be responsible for double replies being made.

**Comment by wc3717** [comment]
We tried a fresh reinstall but we are having the same issue, even with some of the new threads.

**Comment by StephenDaDev** [comment]
@officialpiyush I can not link the changes as they are strictly private, additionally, I don't have them. As Will said, we tried a new install and still seem to be having the issue.

**Comment by officialpiyush** [comment]
@StephenDaDev Private changes in a fully open source bot? ( If you don't want to disclose them in the issue, you can DM me on discord )

**Comment by Forcellrus** [comment]
I have experienced the same issue today, it's the first time we've had it happen since using the bot for over one year now. Setup: bot is hosted on AWS EC2 (Ubuntu Server) and we're running the bot's inbox on a separate Discord server.

**Comment by StephenDaDev** [comment]
Another Sep-Server instance, was the thread left from 24~48 hours?

**Comment by StephenDaDev** [comment]
It has been reported, that the reason for this to happen is the ``thread.recipient`` turns to null
